src/App.js

```diff
@@ -14,11 +15,87 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    isStarted: false,
+    isReversed: false,
+    sortBy: '',
+    goods: goodsFromServer,
+  }
+
+  start = () => {
+    this.setState({ isStarted: true });
+  }
+
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  }
+
+  sortByName = () => {
+    this.setState({ sortBy: 'name' });
+  }
+
+  sortByLength = () => {
+    this.setState({ sortBy: 'length' });
+  }
+
+  reset = () => {
+    this.setState({
+      isReversed: false,
+      sortBy: '',
+    });
+  }
+
+  render() {
+    const { isStarted, isReversed, sortBy, goods } = this.state;
+    const { start, reverse, sortByName, sortByLength, reset } = this;
+    const visibleGoods = [...goods];
+
+    visibleGoods.sort((prevStuff, currentStuff) => {
+      switch (sortBy) {
+        case 'name':
+          return prevStuff.localeCompare(currentStuff);
+
+        case 'length':
+          return prevStuff.length - currentStuff.length;
+
+        default:
+          return 0;
+      }
+    });
+
+    if (isReversed) {
+      visibleGoods.reverse();
+    }
+
+    return (
+      <div className="App">
+        <h1>Goods</h1>
+        {!isStarted
+          ? <button type="button" onClick={start}>Start</button>
```

Comment:
форматирование никто не отменял)
```suggestion
          ? <button
               type="button"
               onClick={start}
             >
               Start
             </button>
```